Userland/Libraries/LibWeb/Fetch/Infrastructure/HTTP/Headers.cpp

```diff
@@ -257,22 +257,38 @@ ErrorOr<Vector<Header>> HeaderList::sort_and_combine() const
 
     // 2. Let names be the result of convert header names to a sorted-lowercase set with all the names of the headers in list.
     Vector<ReadonlyBytes> names_list;
+    TRY(names_list.try_ensure_capacity(size()));
     for (auto const& header : *this)
-        names_list.append(header.name);
+        names_list.unchecked_append(header.name);
     auto names = TRY(convert_header_names_to_a_sorted_lowercase_set(names_list));
 
     // 3. For each name of names:
     for (auto& name : names) {
-        // 1. Let value be the result of getting name from list.
-        // 2. Assert: value is not null.
-        auto value = TRY(get(name)).value();
-
-        // 3. Append (name, value) to headers.
-        auto header = Infrastructure::Header {
-            .name = move(name),
-            .value = move(value),
-        };
-        headers.append(move(header));
+        // 1. If name is `set-cookie`, then:
+        if (name == "set-cookie"sv.bytes()) {
+            // 1. Let values be a list of all values of headers in list whose name is a byte-case-insensitive match for name, in order.
+            // 2. For each value of values:
+            for (auto const& [header_name, value] : *this) {
+                if (StringView { header_name }.equals_ignoring_case(name)) {
+                    // 1. Append (name, value) to headers.
+                    auto header = TRY(Header::from_string_pair(name, value));
+                    TRY(headers.try_append(move(header)));
+                }
+            }
+        }
+        // 2. Otherwise:
+        else {
+            // 1. Let value be the result of getting name from list.
+            // 2. Assert: value is not null.
+            auto value = TRY(get(name)).value();
+
+            // 3. Append (name, value) to headers.
```

Comment:
I guess there is no need to assert value is not null here?

Reply:
Rewrote this slightly to capture the `OptionalByteBuffer`, `VERIFY` it has a value in step 3 and then `release_value()` that into the `Header`. Thanks!